src/CDSAPI.jl

```diff
@@ -1,88 +1,103 @@
 module CDSAPI
 
+using Base64
 using HTTP
 using JSON
-using Base64
+using Logging
 
 export
-    py2ju
+        retrieve, cdskeys
 
 """
-    retrieve(name, params, filename)
-
-Retrieves data for `name` from the Climate Data Store
-with the specified `params` and stores it in the current
-directory as `filename`.
+    retrieve(
+        dataset::AbstractString,
+        params::AbstractDict,
+        fname::AbstractString,
+        keys::AbstractDict = cdskeys()
+    )
+
+Retrieves datasets from the Climate Data Store, with options specified in a Julia Dictionary and saves it into a specified file.
+
+Arguments:
+    * `fname::AbstractString` : string that contains the path and name of the file that the data is to be saved into
+    * `dataset::AbstractString` : string specifies the name of the dataset within the Climate Data Store that the `retrieve` function is attempting to retrieve data from
+    * `params::AbstractDict` : dictionary that contains the keywords that specify the properties (e.g. date, resolution, grid) of the data being retrieved
+    * `keys::AbstractDict` : dictionary that contains API Key information read from the .cdsapirc file in the home directory (optional)
 """
-function retrieve(name, params, filename)
-    creds = Dict()
-    open(joinpath(homedir(),".cdsapirc")) do f
-        for line in readlines(f)
-            key, val = strip.(split(line,':', limit=2))
-            creds[key] = val
-        end
-    end
+function retrieve(
+    dataset::AbstractString,
+    params::AbstractDict,
+    fname::AbstractString,
+    keys::AbstractDict = cdskeys()
+)
 
-    apikey = string("Basic ", base64encode(creds["key"]))
+    @info "$(now()) - Welcome to the Climate Data Store"
```

Comment:
Similarly `now()` is also asking for `Dates.jl` as a dependency. I am not sure these will work without dependencies or not. Please correct me if I am getting it wrong.

Reply:
I forgot about this too.  Gimme a bit, I'll make the changes.